lib/spack/spack/binary_distribution.py

```diff
@@ -817,6 +817,20 @@ def checksum_tarball(file):
     return hasher.hexdigest()
 
 
+def _url_etag_group(url):
+    if isinstance(url, str):
+        parsed_url = urllib.parse.urlparse(url)
+
+    if "http" in parsed_url.scheme:
+        # HTTP mirrors may have host specific etags
+        etag_group = parsed_url.host
+    else:
+        # s3:// and file:// use a consistent ETag method
```

Comment:
Do they? minio, openstack etc have an s3-compatible API. Etags are really only useful if you keep track of the pair `(URL, etag)`.

Reply:
So since we don't have stable tarballs for binaries (yet) this is certainly not true... I will just use the full URL as the key for now.

Reply:
okay, so I did some diggiing, and s3 does have a consistent scheme for identical objects even if they are in a different prefix (same bucket).

I was thinking about the publish mirror in spack-binaries and the way we handle those now is just by copying the `.spack` and `.json.sig` files from the develop mirrors. I did this as a test and the ETag for the original and copy were identical. In the case where they were not, we could have to update the etag in the `.json.sig`, but since they are the same I think it is worth while to keep this logic. If the ETag does end up as different, then that should be a sign that `.json.sig` and and the `.spack` came from a different source.